unit-tests/heitech.zer0mqXt.core.tests/PubSubTests.cs

```diff
@@ -74,6 +72,29 @@ public async Task Subscriber_Cancellation_works()
             
         }
 
+        [Fact]
+        public async Task PubSub_with_retry_works()
+        {
+            // Arrange
+            Message capturedResponse = null;
+            using var socket = ConfigurationTestData.BuildInProcSocketInstanceForTest("retry-socket", timeoutInMs: 500, usePblshr: true);
+            var waitHandle = new ManualResetEvent(false);
+            // request in background thread
+            socket.RegisterSubscriber<Message>(msg => { capturedResponse = msg; waitHandle.Set();});
+            // wait a second for retry
+            Thread.Sleep(250);
+
+            // Act
+            // setup server and wait for retry to work
+            await socket.PublishAsync(new Message { ThisIsAPublishedMessageText = "published-message" });
+
+            // Assert
+            Thread.Sleep(1500);
```

Comment:
I'm not sure if this actually hinders parallel test execution a bit. Perhaps use `await Task.Delay()` instead.
Also in line `85`.

Reply:
Yes, this is also not really what I wanted to do, as the tests become flaky with either solution. Interestingly enough the thread.sleep has the more consistent results on my machine. 

I will look into it again and try to use WaitHandles instead (where possible)